src/main/java/com/google/api/generator/engine/ast/JavaDocComment.java

```diff
@@ -0,0 +1,112 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.api.generator.engine.ast;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@AutoValue
+public abstract class JavaDocComment implements Comment {
+
+  public abstract ImmutableList<String> comments();
+
+  public static Builder builder() {
+    return new AutoValue_JavaDocComment.Builder();
+  }
+
+  public static JavaDocComment withComment(String comment) {
+    return JavaDocComment.builder().addComment(comment).build();
+  }
+
+  @Override
+  public String comment() {
+    return String.join("\n", comments().stream().collect(Collectors.toList()));
+  }
+
+  public void accept(AstNodeVisitor visitor) {
+    visitor.visit(this);
+  }
+
+  @AutoValue.Builder
+  public abstract static class Builder {
+    protected abstract ImmutableList.Builder<String> commentsBuilder();
+
+    public Builder setDeprecated(String deprecatedText) {
+      commentsBuilder().add(String.format("@deprecated %s", deprecatedText));
+      return this;
+    }
+
+    public Builder addComment(String comment) {
+      commentsBuilder().add(comment);
+      return this;
+    }
+
+    public Builder addParam(String name, String description) {
+      String parameter = String.format("@param %s %s", name, description);
+      commentsBuilder().add(parameter);
```

Comment:
Could we please make this consistent with `setDeprecated`  where we add the string directly? Same below.

Reply:
done.